src/debug-integration.adoc

```diff
@@ -185,8 +185,12 @@ If {cheri_default_ext_name} is implemented:
 ** This only needs doing once after resetting the core.
 * The <<m_bit>> is used on debug mode entry to determine which CHERI execution mode to enter.
 
-The <<m_bit>> is the only writeable field in <<dinfc>>.
-Therefore if {cheri_default_ext_name} is not implemented then it is read-write with no writeable fields.
+The <<m_bit>> is the only writeable field in <<dinfc>>:
+
+* It is only updated on a <<CSRRW>> write _if_ the capability being written has a valid <<m_bit>>.
```

Comment:
```suggestion
* It is only updated on a <<CSRRW>> write _if_ the capability being written is an <<infinite-cap>> capability. 
```
Should we add this constraint (which implies valid permissions encoding with M) or do we just ignore all other bits?

Reply:
we now have a new approach, I'll put a comment below